src/sentry_core.c

```diff
@@ -944,6 +946,7 @@ sentry_transaction_finish(sentry_transaction_t *opaque_tx)
     sentry_value_remove_by_key(tx, "op");
     sentry_value_remove_by_key(tx, "description");
     sentry_value_remove_by_key(tx, "status");
+    sentry_value_remove_by_key(tx, "extra");
```

Comment:
the current failing test (in [test_integration_http.py](https://github.com/getsentry/sentry-native/blob/bf14b8533a3b26853e4e6fecf2f955deaa29e2d8/tests/test_integration_http.py#L544-L548)) fails because we remove the "extra" data from the transaction; should we not remove it, or change the test? (also related to the comment above)

Reply:
I'm glad it failed; that is precisely what this test is for. We're changing the contract to the backend. In this case, the backend invited the change, so we must assert whatever the test expected in `extra` to be in `contexts.trace.data`. Changing the test is the right choice here.